src/components/Tokens/TokenDetails/MobileBalanceSummaryFooter.tsx

```diff
@@ -123,7 +121,7 @@ export default function MobileBalanceSummaryFooter({
           </BalanceTotal>
         </BalanceInfo>
       )}
-      <SwapButton to={`/swap?outputCurrency=${outputTokenAddress}`}>
+      <SwapButton to={`/swap?outputCurrency=${tokenAmount?.currency.address}`}>
```

Comment:
I feel like we shouldn't leave the option open for a swap button rendered w/o a valid url. Could do either: 
- don't render w/o a valid url including outputCurrency
- render, but with only `/swap` and no query parameters